src/coreclr/vm/ceeload.cpp

```diff
@@ -337,17 +337,8 @@ IMetaDataEmit *Module::GetValidatedEmitter()
         // verify anything. To ensure we don't break back-compat the verifications are not enabled by default.
         // Right now I have only added verifications for NGEN images, but in the future we might want verifications
         // for all modules.
```

Comment:
We don't use it for anything else currently. I'm a little torn because metadata issues are a common and hard to debug problem for profilers, but this particular implementation didn't do anything super interesting and only was applied to NGEN images. I had been considering getting it working with non NGEN images for a while, but I'm not sure when that would actually happen.

It probably makes sense to just rip out ProfilerMetadataEmitValidator at this point